doc/specs/#500 - Auto generate manifests for newer versions.md

```diff
@@ -0,0 +1,87 @@
+---
+author: Vedant Mohan Goyal @vedantmgoyal9
+created on: 2025-01-25
+last updated: 2025-03-09
+issue id: https://github.com/microsoft/winget-pkgs/issues/500
+---
+
+# Auto-generate Manifests for Newer Versions
+
+## Abstract
+
+This spec outlines the design for automation that generates manifests for new versions of applications in the Windows Package Manager Community Repository (winget-pkgs). It describes a meta-manifest that provides the necessary information (e.g., version, installer URLs, release date, and notes) to automate the creation of new manifests.
+
+## Problem Statement
+
+Currently, winget-pkgs relies on community contributors to update package versions, leading to delays between a new software release and its availability in WinGet. This delay is problematic when updates address security vulnerabilities, leaving users exposed. Faster updates would make WinGet more reliable, particularly for businesses that depend on timely and verified application availability.
+
+## Solution Design
+
+The proposed automation will automatically open pull requests (PRs) for new versions of applications when a new release is detected via the software’s API.
+
+```jsonc
+{
+  "$schema": "<link to schema>",
+  "PackageIdentifier": "AdGuard.AdGuard",
+  "Source": [
+    {
+      "InvokeType": "RestMethod",
+      "Uri": "",
+      "Method": "Get",
+      // other switches and args supported by Invoke-RestMethod/WebRequestHeaders
+      // like Body, UserAgent, MaximumRedirection, SkipHttpErrorCheck, ErrorAction
+    }
+  ],
+  "PostResponseScript": "$Response = $Response.response",
+  "Update": {
+    "PackageVersion": "$Response.version.'#cdata-section'",
+    "InstallerUrls": "$Response.'update-url'.'#cdata-section'"
+  },
+  "AdditionalInfo": {
+    "ForceUpgrade": false
+  },
+  "Skip": {
+    "Skip?": false,
+    "Reason": null
+  }
+}
+```
+
+The above JSON defines automation metadata and will be stored in a `.auto_updates.json` file alongside `.validation` files.
```

Comment:
It would probably be better to have a separate directory structure specifically for auto-updates instead of including metadata in the `manifests` folder. While having the data right next to all of the manifests makes it easy for users to find, it greatly increases the complexity cost of a solution since whatever is built will have to recursively filter through all the directories to find the metadata files instead of filtering to only a known sub-tree